packages/api/src/resolvers/function_resolvers.ts

```diff
@@ -386,6 +386,47 @@ export const functionResolvers = {
         ? ContentReader.Pdf
         : ContentReader.Web
     },
+    async readingProgressPercent(
+      article: { id: string; readingProgressPercent?: number },
+      _: unknown,
+      ctx: WithDataSourcesContext & { claims: Claims }
+    ) {
+      // != used here to check for null or undefined
+      if (article.readingProgressPercent != null) {
+        return article.readingProgressPercent
+      }
+      console.log(
+        'looking up reading progress for article',
+        article.id,
+        article
+      )
+      return (
+        (
+          await getPageByParam({
+            userId: ctx.claims.uid,
+            _id: article.id,
+          })
+        )?.readingProgressPercent || 0
+      )
+    },
+    async readingProgressAnchorIndex(
+      article: { id: string; readingProgressAnchorIndex?: number },
+      _: unknown,
+      ctx: WithDataSourcesContext & { claims: Claims }
+    ) {
+      // != used here to check for null or undefined
+      if (article.readingProgressAnchorIndex != null) {
+        return article.readingProgressAnchorIndex
+      }
+      return (
+        (
+          await getPageByParam({
```

Comment:
@sywhb it looks like we are failing this lookup for a newsletter I just had saved to my account, when I look in elastic the newsletter as an empty id.

```
{
        "sort": [
          1647460684897
        ],
        "_type": "_doc",
        "_source": {
          "hash": "b5b0ce86519db644057aa9e513fb03ae",
          "description": "Start and end your work day with the stories that matter from Axios by Mike Allen, the world’s most-wired reporter. Then finish the night with what matters -- and lasts -- in life with insights from Mike, Jim VandeHei and Erica Pandey.",
          "author": "Mike Allen",
          "url": "https://link.axios.com/view/61eef5e5492d41078a32440cg3s4z.akaa/ae9285ff",
          "image": "https://static.axios.com/img-email/socialcard_axiosam3.jpg",
          "title": "Axios Finish Line",
          "userId": "e7ad07f2-8805-11eb-8d0b-53d13fcb2ac1",
          "slug": "axios-pm-the-great-communicator-17f944f0ebb",
          "updatedAt": "2022-03-16T19:58:04.897Z",
          "pageType": "UNKNOWN",
          "id": "",
          "createdAt": "2022-03-16T19:58:04.897Z"
        },
        "_score": null,
        "_index": "pages",
        "_ignored": [
          "originalHtml.keyword"
        ],
        "_id": "mC1PlH8BNO0b99P4FMq0"
      },
      ```